### ISSUE TITLE
Discarded report from unrecognized worker

### ISSUE THREAD

**Issue by michie1** [locustio/locust#3003]
### Prerequisites

- [X] I am using [the latest version of Locust](https://github.com/locustio/locust/releases/)
- [X] I am reporting a bug, not asking a question

### Description

This is a follow up of https://github.com/locustio/locust/issues/3000. The UI now shows the correct number, but only when one worker node is used. When adding `--processes 10` and specifying 10 users in the UI I still get a lower number in the UI compared with what the nodejs command is showing.

The logs is saying "locust.runners: ... reported that is has stopped" and later "Discarded report from unrecognized worker ...". This is logged multiple times for multiples runners.

```
const http = require('http');

let requestCount = 1;

const server = http.createServer((req, res) => {
  setTimeout(() => {
    console.log(requestCount);
    res.writeHead(200, { 'Content-Type': 'text/plain' });
    res.end(`Request count: ${requestCount}`);
    requestCount++; // Increment the request counter
  }, 1000);
});

const PORT = 3000;
server.listen(PORT, () => {
    console.log(`Server running at http://localhost:${PORT}`);
});
```

### Command line

locust -s 10 --processes 10

### Locustfile contents

```python3
from locust import HttpUser, task

i = 0

class HelloWorldUser(HttpUser):
    @task
    def hello_world(self):
        global i
        print("before", i)
        self.client.get("/")
        print("after", i)
        i += 1
```
```


### Python version

3.10

### Locust version

2.32.4

### Operating system

Ubuntu 22.04.5 WSL

**Comment by michie1** [comment]
Same holds when I remove the `setTimeout` (I thought maybe that would introduce count differences) and specify the Runtime to 10 seconds instead of clicking on the "Stop" button.

**Comment by cyberw** [comment]
I can reproduce this as well. I wonder when it broke... Anyways, I'm on it...

**Comment by cyberw** [comment]
Actually, I'm only able to reproduce it _with_ `--stop-timeout`. I still sometimes get the log message at shutdown without -s, but then no requests are actually missing. So still a bug, but much smaller scope...

**Comment by michie1** [comment]
Please remove stale label.

**Comment by cyberw** [comment]
Can you reproduce this consistently with nothing in special in your locustfile? Are you using `--stop-timeout`? Are you using docker?

Worker restart on stop test is weird and something we wanna fix, but it probably unrelated.

**Comment by frankjkelly** [comment]
@cyberw Yes - see this consistently. I guess one thing that is special is we are using the `SocketIOUser` https://github.com/SvenskaSpel/locust-plugins/blob/master/locust_plugins/users/socketio.py 
(A slightly modified one but nothing magical to my newbie eye at least)

We are deployed on Kubernetes with a Master and a Worker.

Dockerfile below
```
# Example from https://github.com/bneijt/poetry-lock-docker/
FROM python:3.10.16-slim

WORKDIR /project

COPY dist/*.whl /project/

ARG PIP_INDEX_URL

# Needed for fastavro
RUN apt-get update 
RUN apt-get -y install gcc

RUN pip install --no-cache-dir *.whl  \
    && rm -rf *.whl

# Temp fix for CVE-2021-46848
RUN apt-get --only-upgrade install -y libexpat1 libtasn1-6

USER 1001
COPY locust/tasks/ /locust-tasks/

#From https://github.com/locustio/locust/blob/master/Dockerfile
EXPOSE 8089 5557
# turn off python output buffering
ENV PYTHONUNBUFFERED=1
ENTRYPOINT ["locust"]
```
